modules/core/src/v2/coins/algo.ts

```diff
@@ -230,53 +234,41 @@ export class Algo extends BaseCoin {
    */
   explainTransaction(
     params: ExplainTransactionOptions,
-    callback?: NodeCallback<TransactionExplanation>
-  ): Bluebird<TransactionExplanation> {
-    return co<TransactionExplanation>(function* () {
-      // take txHex first always, but as it might already be signed, take halfSigned second
+    callback?: NodeCallback<AlgoTransactionExplanation>
+  ): Bluebird<AlgoTransactionExplanation> {
+    return co<AlgoTransactionExplanation>(function* () {
       const txHex = params.txHex || (params.halfSigned && params.halfSigned.txHex);
-
-      if (!txHex) {
-        throw new Error('missing required param txHex or halfSigned.txHex');
+      if (!txHex || !params.feeInfo) {
+        throw new Error('missing explain tx parameters');
       }
-      let tx, type, senderAddress, voteKeyBase64, voteLastBlock;
-      try {
-        const txToHex = Buffer.from(txHex, 'base64');
-        const decodedTx = Encoding.decode(txToHex);
-
-        // if we are a signed msig tx, the structure actually has the { msig, txn } as the root object
-        // if we are not signed, the decoded tx is the txn - refer to partialSignTxn and MultiSig constructor
-        //   in algosdk for more information
-        const txnForDecoding = decodedTx.txn || decodedTx;
-        if (!!txnForDecoding.votekey) {
-          type = txnForDecoding.type;
-          senderAddress = Address.encode(txnForDecoding.snd);
-          voteKeyBase64 = txnForDecoding.votekey.toString('base64');
-          voteLastBlock = txnForDecoding.votelst;
-        }
-
-        tx = Multisig.MultiSigTransaction.from_obj_for_encoding(txnForDecoding);
-      } catch (ex) {
-        throw new Error('txHex needs to be a valid tx encoded as base64 string');
-      }
-
-      const id = tx.txID();
-      const fee = { fee: tx.fee };
 
-      const outputAmount = tx.amount || 0;
-      const outputs: { amount: number; address: string }[] = [];
-      if (tx.to) {
-        outputs.push({
-          amount: outputAmount,
-          address: Address.encode(new Uint8Array(tx.to.publicKey)),
-        });
-      }
-
-      // TODO(CT-480): add recieving address display here
-      const memo = tx.note;
+      const factory = accountLib.getBuilder(
+        this.getBaseChain()
+      ) as unknown as accountLib.Algo.TransactionBuilderFactory;
+
+      const txBuilder = factory.from(txHex);
+      const tx = (yield txBuilder.build()) as any;
```

Comment:
nit: can we avoid the cast to `any` here?

Reply:
We can't. It's neccessary to make the next steps.

Reply:
I don't think that's true, even if it's a partial type it's better than `any`